spotless-changelog-lib/src/main/java/com/diffplug/spotless/changelog/GitActions.java

```diff
@@ -118,10 +122,25 @@ private void push(Ref ref, RemoteRefUpdate.Status expected) throws GitAPIExcepti
 	}
 
 	private void push(Consumer<PushCommand> cmd, RemoteRefUpdate.Status expected) throws GitAPIException {
-		PushCommand push = git.push().setCredentialsProvider(creds()).setRemote(cfg.remote);
+		String remoteUrl = git.getRepository().getConfig().getString(ConfigConstants.CONFIG_REMOTE_SECTION, cfg.remote, ConfigConstants.CONFIG_KEY_URL);
+
+		PushCommand push = git.push().setRemote(cfg.remote);
+		if (remoteUrl.startsWith("http://") || remoteUrl.startsWith("https://")) {
+			push = push.setCredentialsProvider(creds());
+		} else if (remoteUrl.startsWith("ssh://")) {
+			push.setTransportConfigCallback(transport -> {
+				SshTransport sshTransport = (SshTransport) transport;
+				sshTransport.setSshSessionFactory(new JschConfigSessionFactory() {
+					@Override
+					protected void configure(OpenSshConfig.Host host, Session session) {
+						session.setConfig("StrictHostKeyChecking", "no");
```

Comment:
This is a pretty risky default. What do you think about adding a flag like `--sshDisableStrictHostKeyChecking`, and printing an error on failure that says "try --sshDisableStrictHostKeyChecking if you're sure that it's the right repo" or something like that?

Reply:
i think the idea with the flag is very good. I can still install the flag with an error message if the ssh check fails. I would then also adjust the extension (gitcfg). 
Parameter: -DsshDisableStrictHostKeyChecking
Extension: sshDisableStrictHostKeyChecking = true

Reply:
sounds great